docs/Delivery-Practices/Build-and-Release/Engineering/Quality-Standards/React-Best-Practises.md

```diff
@@ -0,0 +1,39 @@
+
+## Guidelines for Writing React Code in TypeScript
+
+1. Use semantic HTML: Use semantic HTML tags that describe the content of the page rather than just the appearance. This will help with accessibility and search engine optimization, and make your code easier to read and maintain.
+
+1. Use React Testing Library: Use React Testing Library to write unit and integration tests for your components. This library provides a simple and intuitive API for testing React components and encourages developers to write tests that reflect the way users interact with the application.
+
+1. Use the Atomic Design Methodology: The Atomic Design Methodology is a system for designing and building UI components in a structured and scalable way. By breaking down your UI into smaller, reusable components, you can create a more flexible and maintainable codebase.
+
+1. Use React Query: React Query is a library for managing data fetching and caching in React applications. It provides a simple and intuitive API for fetching data from APIs, and supports caching, pagination, and other advanced features.
+
+1. Use strict mode: Use the React strict mode to enable additional checks and warnings for potential problems in your code. This can help you catch potential issues early and improve the overall quality of your code.
+
+1. Use ESLint and Prettier: Use ESLint and Prettier to enforce coding standards and formatting rules in your codebase. These tools can help you maintain consistency and improve the readability of your code.
+
+1. Use CSS modules or styled-components: Use CSS modules or styled-components to manage the styling of your components. These tools provide a way to write scoped styles that are only applied to specific components and can help you avoid naming collisions and other styling issues.
+
+1. Use prop drilling sparingly: Avoid excessive prop drilling by using React Query or React Context. These provide a centralized store for managing the state of your application and can help you avoid passing props down through multiple levels of components.
+
+1. Use async/await for data fetching: Use async/await syntax to fetch data asynchronously from APIs or other sources. This can help you avoid callback hell and make your code easier to read and maintain.
+
+1. Use hooks effectively: Use React hooks like useState and useEffect to manage state and side effects in your components. Hooks provide a simple and intuitive API for managing state and can help you avoid class components and the complexities that come with them.
```

Comment:
I'd also highlight to use them as sparingly as possible. 

1. useState
  - Ensure we are using "derived state" wherever possible and not "state duplication"
    - https://blog.webdevsimplified.com/2019-11/never-store-derived-state/
  - useState causes the DOM to rerender. Only use useState if we want rerenders to occur. If not, you may want `useRef`

1. useEffect
  - Used for syncing data; however, this can also cause a derived state issue again